### PR TITLE
feat(build-config): add production build config (RC) (AR-2952) (#1389)

### PR DESCRIPTION
**Pull request by MohamadJaara**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2952" title="AR-2952" target="_blank">StoryAR-2952</a>  Create production build 🚀
  </td></table>
  <br />
 

* feat(build-config): add production build config (RC)

* hide mls options in debug screen when mls is disabled

* apply icon and dependency filtering

* disable minify

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1677/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1677/2/) **failed**.

File: kalium
Comment (Garzas) [inline kalium:1]: Are those changes from kalium needed?

**Review by Garzas — APPROVED** [review]
🥇 just one question about kalium changes

**Review by mchenani — APPROVED** [review]

File: kalium
Comment (MohamadJaara) [reply kalium:1]: it include the following commit 
[chore: improve generic failure logging (](https://github.com/wireapp/kalium/commit/77afec01422167fff4e470782c69c5bfd116b4a0)https://github.com/wireapp/kalium/pull/1638[)](https://github.com/wireapp/kalium/commit/77afec01422167fff4e470782c69c5bfd116b4a0)

File: app/src/beta/kotlin/com/wire/android/ExternalLoggerManager.kt
Comment (vitorhugods) [inline app/src/beta/kotlin/com/wire/android/ExternalLoggerManager.kt:1]: I won't try to stop this train, but we should really unify the `ExternalLoggerManager` into a single file.

Creating a new source set, like "NotProduction", and have `beta`, `dev`, and `internal` depend on it. So all the logic can be put in a single place.

File: kalium
Comment (vitorhugods) [reply kalium:1]: It adds the "logging improvement" PR that I did:

- https://github.com/wireapp/kalium/pull/1638

Actually makes sense to bump this into prod, so we get better logs if people share them with us.

File: default.json [hunk 1]
```diff
@@ -71,5 +71,7 @@
     "update_app_url": "https://wire.com/en/download/",
     "guest_room_link": true,
     "wipe_on_cookie_invalid": false,
-    "wipe_on_device_removal": false
+    "wipe_on_device_removal": false,
+    "report_bug_menu_item_enabled": true,
+    "debug_screen_enabled": true
```
Comment (vitorhugods) [inline default.json:76]: Should `debug_screen_enabled` be `false` for `prod`? I'm not sure TBH

**Review by vitorhugods — COMMENTED** [review]
Looks good to me. Just a couple of comments, really.

Please confirm if the debug screen should be visible on prod builds.

File: default.json (hunk 1 above)
Comment (MohamadJaara) [reply default.json:76]: i was true in 4.0.4

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1677/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6251705)(6251705)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6251705)(6251705)-staging-compat.apk)

**Pull request merged**